src/tornado/util.cljc

```diff
@@ -246,13 +254,54 @@
        (into {})
        not-empty))
 
-(def ^:dynamic *compress?*
-  "Moved from this ns to util in version 0.2.10 to prevent cyclic dependency needed in `ns-kw->str`."
-  false)
-
 (defn ns-kw->str [expr]
   (if (and (keyword? expr)
            (namespace expr))
-    (-> (str (namespace expr) (if *compress?* "-" "--") (name expr))
+    (-> (str (namespace expr) (if ctx/*compress?* "-" "--") (name expr))
         (str/replace #"\." "-"))
-    (name expr)))
\ No newline at end of file
+    (name expr)))
+
+(defn or-nil [x]
+  (if (nil? x) "nil" x))
+
+(defn update-in-keys
+  "Given a map or a record, a function, a common partial path and keys which will be
+  appended to that path, updates all keys in the given map with that function."
+  [m f path & ks]
+  (reduce (fn [m k]
+            (update-in m (conj path k) f))
+          m
+          ks))
+
+(defn char-at
+  [s n]
+  #?(:clj  (.charAt ^String s ^int n)
+     :cljs (.at s n)))
+
+(defn normalize
+  "Normalizes a float 0 <= x <= 255 to 0 <= x <= 1."
+  [x]
+  (let [v (double (/ x 255))]
+    (if (= (double (int v)) v)
+      (int v)
+      v)))
+
+(defn denormalize
+  "Denormalizes a float 0 <= x <= 1 to 0 <= x <= 255."
+  [x]
+  (math-round (* x 255)))
+
+(defn partition-string
```

Comment:
this might be easier to write by turning the string to a seq of characters, using the actual partition method and then mapping str over the output of it.